### ISSUE TITLE
Leak in BinaryMemcacheDecoderTest

### ISSUE THREAD

**Issue by Scottmitch** [netty/netty#4169]
```
Running io.netty.handler.codec.memcache.binary.BinaryMemcacheDecoderTest
16:47:36.889 [main] ERROR io.netty.util.ResourceLeakDetector - LEAK: ByteBuf.release() was not called before it's garbage-collected. See http://netty.io/wiki/reference-counted-objects.html for more information.
Recent access records: 5
#5:
	io.netty.buffer.AdvancedLeakAwareByteBuf.release(AdvancedLeakAwareByteBuf.java:722)
	io.netty.util.ReferenceCountUtil.release(ReferenceCountUtil.java:84)
	io.netty.util.ReferenceCountUtil.safeRelease(ReferenceCountUtil.java:109)
	io.netty.channel.ChannelOutboundBuffer.remove(ChannelOutboundBuffer.java:261)
	io.netty.channel.embedded.EmbeddedChannel.doWrite(EmbeddedChannel.java:385)
	io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:815)
	io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:782)
	io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1234)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeFlushNow(ChannelHandlerInvokerUtil.java:165)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeFlush(EmbeddedEventLoop.java:242)
	io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:272)
	io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:115)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeFlushNow(ChannelHandlerInvokerUtil.java:165)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeFlush(EmbeddedEventLoop.java:242)
	io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:272)
	io.netty.channel.DefaultChannelPipeline.flush(DefaultChannelPipeline.java:997)
	io.netty.channel.AbstractChannel.flush(AbstractChannel.java:226)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:220)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
#4:
	io.netty.buffer.AdvancedLeakAwareByteBuf.retain(AdvancedLeakAwareByteBuf.java:694)
	io.netty.buffer.AdvancedLeakAwareByteBuf.retain(AdvancedLeakAwareByteBuf.java:31)
	io.netty.util.ReferenceCountUtil.retain(ReferenceCountUtil.java:36)
	io.netty.channel.embedded.EmbeddedChannel.doWrite(EmbeddedChannel.java:383)
	io.netty.channel.AbstractChannel$AbstractUnsafe.flush0(AbstractChannel.java:815)
	io.netty.channel.AbstractChannel$AbstractUnsafe.flush(AbstractChannel.java:782)
	io.netty.channel.DefaultChannelPipeline$HeadContext.flush(DefaultChannelPipeline.java:1234)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeFlushNow(ChannelHandlerInvokerUtil.java:165)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeFlush(EmbeddedEventLoop.java:242)
	io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:272)
	io.netty.channel.ChannelOutboundHandlerAdapter.flush(ChannelOutboundHandlerAdapter.java:115)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeFlushNow(ChannelHandlerInvokerUtil.java:165)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeFlush(EmbeddedEventLoop.java:242)
	io.netty.channel.AbstractChannelHandlerContext.flush(AbstractChannelHandlerContext.java:272)
	io.netty.channel.DefaultChannelPipeline.flush(DefaultChannelPipeline.java:997)
	io.netty.channel.AbstractChannel.flush(AbstractChannel.java:226)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:220)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
#3:
	Hint: 'DefaultChannelPipeline$HeadContext#0' will handle the message from this point.
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:264)
	io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:112)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeWriteNow(ChannelHandlerInvokerUtil.java:157)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeWrite(EmbeddedEventLoop.java:237)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:265)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:258)
	io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1039)
	io.netty.channel.AbstractChannel.write(AbstractChannel.java:268)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:217)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
#2:
	io.netty.buffer.AdvancedLeakAwareByteBuf.writeLong(AdvancedLeakAwareByteBuf.java:502)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder.encodeHeader(BinaryMemcacheRequestEncoder.java:36)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder.encodeHeader(BinaryMemcacheRequestEncoder.java:23)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:39)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:27)
	io.netty.handler.codec.memcache.AbstractMemcacheObjectEncoder.encode(AbstractMemcacheObjectEncoder.java:47)
	io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:89)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeWriteNow(ChannelHandlerInvokerUtil.java:157)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeWrite(EmbeddedEventLoop.java:237)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:265)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:258)
	io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1039)
	io.netty.channel.AbstractChannel.write(AbstractChannel.java:268)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:217)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
#1:
	io.netty.buffer.AdvancedLeakAwareByteBuf.writeInt(AdvancedLeakAwareByteBuf.java:496)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder.encodeHeader(BinaryMemcacheRequestEncoder.java:35)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheRequestEncoder.encodeHeader(BinaryMemcacheRequestEncoder.java:23)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:39)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:27)
	io.netty.handler.codec.memcache.AbstractMemcacheObjectEncoder.encode(AbstractMemcacheObjectEncoder.java:47)
	io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:89)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeWriteNow(ChannelHandlerInvokerUtil.java:157)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeWrite(EmbeddedEventLoop.java:237)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:265)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:258)
	io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1039)
	io.netty.channel.AbstractChannel.write(AbstractChannel.java:268)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:217)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Created at:
	io.netty.buffer.PooledByteBufAllocator.newDirectBuffer(PooledByteBufAllocator.java:271)
	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:157)
	io.netty.buffer.AbstractByteBufAllocator.directBuffer(AbstractByteBufAllocator.java:148)
	io.netty.buffer.AbstractByteBufAllocator.buffer(AbstractByteBufAllocator.java:85)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:37)
	io.netty.handler.codec.memcache.binary.AbstractBinaryMemcacheEncoder.encodeMessage(AbstractBinaryMemcacheEncoder.java:27)
	io.netty.handler.codec.memcache.AbstractMemcacheObjectEncoder.encode(AbstractMemcacheObjectEncoder.java:47)
	io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:89)
	io.netty.channel.ChannelHandlerInvokerUtil.invokeWriteNow(ChannelHandlerInvokerUtil.java:157)
	io.netty.channel.embedded.EmbeddedEventLoop.invokeWrite(EmbeddedEventLoop.java:237)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:265)
	io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:258)
	io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1039)
	io.netty.channel.AbstractChannel.write(AbstractChannel.java:268)
	io.netty.channel.embedded.EmbeddedChannel.writeOutbound(EmbeddedChannel.java:217)
	io.netty.handler.codec.memcache.binary.BinaryMemcacheEncoderTest.shouldEncodeContent(BinaryMemcacheEncoderTest.java:130)
	sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	java.lang.reflect.Method.invoke(Method.java:497)
	org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:264)
	org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:124)
	org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.018 sec - in io.netty.handler.codec.memcache.binary.BinaryMemcacheDecoderTest
```

**Comment by Scottmitch** [comment]
@normanmaurer - I discovered this leak while testing my fix to https://github.com/netty/netty/issues/4150.  @trustin - Do the build servers still consider a "leaky" build a failed build?

**Comment by trustin** [comment]
@Scottmitch Only for pull requests I guess. We could add a dedicated job for leak checks, but maybe PR jobs should find them? (piggyback ftw)

**Comment by trustin** [comment]
Fixed in e556958abc61f487ae2d3a7f1d813f429e9f0ccc (4.1) and 52bec9726ff89a5d1223de4fa9474f1222f1a609 (master)

**Comment by Scottmitch** [comment]
Thanks for taking care @trustin!

@trustin - PR builds treating leaks as build failures should be sufficient. Although it would be *nice* to have a relatively infrequent spot check leak build on the release branches just incase something slips by.
